NitroxClient/Communication/MultiplayerSession/ConnectionState/AwaitingReservationCredentials.cs

```diff
@@ -15,8 +15,8 @@ public override void NegotiateReservation(IMultiplayerSessionConnectionContext s
             try
             {
                 ValidateState(sessionConnectionContext);
-
-                string reservationCorrelationId = Guid.NewGuid().ToString();
+                ulong steamId = Steamworks.SteamUser.GetSteamID().m_SteamID;
+                string reservationCorrelationId = steamId.ToString();
```

Comment:
Are there any edge cases that occur when the playerId and the reservation id are the same? For example, what if I am authenticating, cancel, then try to authenticate again.  

What was the original motivation for having these two ids be the same?